samples/layouts/dock-manager/add-content-runtime/src/index.ts

```diff
@@ -0,0 +1,219 @@
+                                        }
+                                    ]
+                                }
+                            },
+                            {
+                                type: IgcDockManagerPaneType.contentPane,
+                                contentId: 'content5',
+                                header: 'Unpinned Pane 2',
+                                isPinned: false
+                            }
+                        ]
+                    },
+                    {
+                        type: IgcDockManagerPaneType.splitPane,
+                        orientation: IgcSplitPaneOrientation.vertical,
+                        panes: [
+                            {
+                                type: IgcDockManagerPaneType.tabGroupPane,
+                                size: 200,
+                                panes: [
+                                    {
+                                        type: IgcDockManagerPaneType.contentPane,
+                                        contentId: 'content6',
+                                        header: 'Tab 1'
+                                    },
+                                    {
+                                        type: IgcDockManagerPaneType.contentPane,
+                                        contentId: 'content7',
+                                        header: 'Tab 2'
+                                    },
+                                ]
+                            },
+                            {
+                                type: IgcDockManagerPaneType.contentPane,
+                                contentId: 'content8',
+                                header: 'Content Pane 2'
+                            }
+                        ]
+                    }
+                ]
+            },
+            floatingPanes: [
+                {
+                    type: IgcDockManagerPaneType.splitPane,
+                    orientation: IgcSplitPaneOrientation.horizontal,
+                    floatingHeight: 150,
+                    floatingWidth: 250,
+                    floatingLocation: { x: 300, y: 200 },
+                    panes: [
+                        {
+                            type: IgcDockManagerPaneType.contentPane,
+                            contentId: 'content9',
+                            header: 'Floating Pane'
+                        },
+                    ]
+                },
+            ]
+        };
+
+        addContentPaneBtn!.addEventListener('click', () => {
+            const cp: IgcContentPane = {
+              type: IgcDockManagerPaneType.contentPane,
+              contentId: `newContent${this._counter}`,
+              header: `New Pane ${this._counter}`,
+            };
+
+            this.dockManager.layout.rootPane.panes.push(cp);
+        
+            this.attachPane();
+        })
+
+        addTabGroupPaneBtn!.addEventListener('click', () => {
```

Comment:
If I dock the predefined document tabs one above the other for example and then click the **Add Document** button nothing happens.
In order to avoid the complex code of adding a document I would suggest to remove this button from the sample.

Reply:
I'd rather have a check if a tabGroupPane exists already and add it there.